fullscreen.bs

```diff
@@ -289,14 +301,17 @@ these steps:
 return true if the <a>context object</a> is <a>allowed to use</a> the feature indicated by attribute
 name <code>allowfullscreen</code> and <a>fullscreen is supported</a>, and false otherwise.
 
-<p>The <dfn attribute for=Document><code>fullscreenElement</code></dfn> attribute's getter must
-return <a>context object</a>'s <a>fullscreen element</a>.
-
 <p>The <dfn attribute for=Document><code>fullscreen</code></dfn> attribute's getter must return
 false if <a>context object</a>'s <a>fullscreen element</a> is null, and true otherwise.
 
 <p class=note>Use
-<a attribute for=Document lt=fullscreenElement><code>document.fullscreenElement</code></a> instead.
+<a attribute for=DocumentOrShadowRoot lt=fullscreenElement><code>document.fullscreenElement</code></a>
+instead.
+
+<p>The <dfn attribute for=DocumentOrShadowRoot><code>fullscreenElement</code></dfn> attribute's
+getter must return the result of <a>retargeting</a> <a>context object</a>'s
```

Comment:
You're probably right that having the concept return the topmost element with the flag set in the top layer makes things easier.

I think there's no disagreement about what `fullscreenElement` should return, so the question is what should happen to the other places that use [fullscreen element](https://fullscreen.spec.whatwg.org/#fullscreen-element). Those are:
 * [fully exit fullscreen](https://fullscreen.spec.whatwg.org/#fully-exit-fullscreen), which I *think* can continue to be a document-only concept.
 * The removing steps, where I think having the "unfullscreen element" steps traverse all shadow trees is actually redundant, because the removing steps will be run for those nodes too?
 * In [`requestFullscreen()`](https://fullscreen.spec.whatwg.org/#dom-element-requestfullscreen), where it's just used to avoid an unnecessary event.
 * In [exit fullscreen](https://fullscreen.spec.whatwg.org/#exit-fullscreen), where some tweak are probably needed if you make exit fullscreen per-DocumentOrShadowRoot instead of per-Document.

I'll wait to see the next iteration, thanks for working through this!